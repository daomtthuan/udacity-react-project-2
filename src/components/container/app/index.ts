import AppContainer from './_.view';

export default AppContainer;
