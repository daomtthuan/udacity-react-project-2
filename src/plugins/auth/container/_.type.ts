import { PropsWithChildren } from 'react';

export type AuthContainerProps = PropsWithChildren;
