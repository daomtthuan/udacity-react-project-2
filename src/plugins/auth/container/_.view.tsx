import { Location, Navigate, useLocation } from 'react-router-dom';
import { useAppSelector } from '~plugins/store';

import { AuthContainerProps } from './_.type';

const pathConfig = {
  signIn: '/',
  home: '/home',

  guest: ['/', '/sign-up'],
};

export default function AuthContainer({ children }: Readonly<AuthContainerProps>) {
  const location: Location<string | null> = useLocation();

  const authUser = useAppSelector((state) => state.auth.user);

  // Not authenticated
  if (!authUser) {
    // Location is not guest routes
    if (!pathConfig.guest.includes(location.pathname)) {
      return <Navigate to={pathConfig.signIn} state={location.pathname} />;
    }

    return children;
  }

  // Authenticated
  else {
    // Location has state
    if (location.state) {
      return <Navigate to={location.state} state={null} />;
    }

    // Location is guest routes
    if (pathConfig.guest.includes(location.pathname)) {
      return <Navigate to={pathConfig.home} state={null} />;
    }

    return children;
  }
}
