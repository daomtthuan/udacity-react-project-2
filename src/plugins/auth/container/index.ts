import AuthContainer from './_.view';

export default AuthContainer;
