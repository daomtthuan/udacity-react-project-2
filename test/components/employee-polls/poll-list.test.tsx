import PollList, {
  PollListProps,
} from '~components/feature/employee-polls/poll-list';
import { TestContainer } from '~test/__mocks__/container';

import { render } from '@testing-library/react';

describe('PollList renders correctly', () => {
  test.each<PollListProps>([
    {
      items: [],
    },

    {
      items: [
        {
          id: 'thuandmt',
          password: '123',
          name: 'Thuan Dao',
          avatarURL: '/images/avatar/7.png',
          answers: {
            'xj352vofupe1dqz9emx13r': 'optionOne',
            'vthrdm985a262al8qx3do': 'optionTwo',
            'am8ehyc8byjqgar0jgpub9': 'optionOne',
            '6ni6ok3ym7mf1p33lnez': 'optionTwo',
            '8xf0y6ziyjabvozdd253nd': 'optionOne',
            'a0xo5jhg6y7tj2hwt3t6k': 'optionTwo',
          },
          questions: [],
          answerCount: 6,
        },
        {
          id: 'sarahedo',
          password: 'password123',
          name: 'Sarah Edo',
          avatarURL: '/images/avatar/1.png',
          answers: {
            '8xf0y6ziyjabvozdd253nd': 'optionOne',
            '6ni6ok3ym7mf1p33lnez': 'optionOne',
            'am8ehyc8byjqgar0jgpub9': 'optionTwo',
            'loxhs1bqm25b708cmbf3g': 'optionTwo',
          },
          questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9'],
          answerCount: 4,
        },
        {
          id: 'mtsamis',
          password: 'xyz123',
          name: 'Mike Tsamis',
          avatarURL: '/images/avatar/3.png',
          answers: {
            'xj352vofupe1dqz9emx13r': 'optionOne',
            'vthrdm985a262al8qx3do': 'optionTwo',
            '6ni6ok3ym7mf1p33lnez': 'optionOne',
          },
          questions: ['6ni6ok3ym7mf1p33lnez', 'xj352vofupe1dqz9emx13r'],
          answerCount: 3,
        },
        {
          id: 'tylermcginnis',
          password: 'abc321',
          name: 'Tyler McGinnis',
          avatarURL: '/images/avatar/2.png',
          answers: {
            vthrdm985a262al8qx3do: 'optionOne',
            xj352vofupe1dqz9emx13r: 'optionTwo',
          },
          questions: ['loxhs1bqm25b708cmbf3g', 'vthrdm985a262al8qx3do'],
          answerCount: 2,
        },
      ],
    },
  ])('PollList render as expected with props %o', (props) => {
    const {baseElement} = render(
      <TestContainer>
        <PollList {...props} />
      </TestContainer>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
