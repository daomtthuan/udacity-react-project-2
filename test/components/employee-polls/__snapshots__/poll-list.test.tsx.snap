// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollList renders correctly PollList render as expected with props {
  items: [
    {
      id: 'thuandmt',
      password: '123',
      name: 'Thuan Dao',
      avatarURL: '/images/avatar/7.png',
      answers: {
        xj352vofupe1dqz9emx13r: 'optionOne',
        vthrdm985a262al8qx3do: 'optionTwo',
        am8ehyc8byjqgar0jgpub9: 'optionOne',
        '6ni6ok3ym7mf1p33lnez': 'optionTwo',
        '8xf0y6ziyjabvozdd253nd': 'optionOne',
        a0xo5jhg6y7tj2hwt3t6k: 'optionTwo'
      },
      questions: [ [length]: 0 ],
      answerCount: 6
    },
    {
      id: 'sarahedo',
      password: 'password123',
      name: 'Sarah Edo',
      avatarURL: '/images/avatar/1.png',
      answers: {
        '8xf0y6ziyjabvozdd253nd': 'optionOne',
        '6ni6ok3ym7mf1p33lnez': 'optionOne',
        am8ehyc8byjqgar0jgpub9: 'optionTwo',
        loxhs1bqm25b708cmbf3g: 'optionTwo'
      },
      questions: [
        '8xf0y6ziyjabvozdd253nd',
        'am8ehyc8byjqgar0jgpub9',
        [length]: 2
      ],
      answerCount: 4
    },
    {
      id: 'mtsamis',
      password: 'xyz123',
      name: 'Mike Tsamis',
      avatarURL: '/images/avatar/3.png',
      answers: {
        xj352vofupe1dqz9emx13r: 'optionOne',
        vthrdm985a262al8qx3do: 'optionTwo',
        '6ni6ok3ym7mf1p33lnez': 'optionOne'
      },
      questions: [ '6ni6ok3ym7mf1p33lnez', 'xj352vofupe1dqz9emx13r', [length]: 2 ],
      answerCount: 3
    },
    {
      id: 'tylermcginnis',
      password: 'abc321',
      name: 'Tyler McGinnis',
      avatarURL: '/images/avatar/2.png',
      answers: {
        vthrdm985a262al8qx3do: 'optionOne',
        xj352vofupe1dqz9emx13r: 'optionTwo'
      },
      questions: [ 'loxhs1bqm25b708cmbf3g', 'vthrdm985a262al8qx3do', [length]: 2 ],
      answerCount: 2
    },
    [length]: 4
  ]
} 1`] = `
<body>
  <div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover table-bordered align-middle caption-top"
      >
        <caption>
          List of users
        </caption>
        <thead
          class="table-light"
        >
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              User
            </th>
            <th
              scope="col"
            >
              Answered
            </th>
            <th
              scope="col"
            >
              Asked
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              1
            </th>
            <td>
              <div
                class="d-flex align-items-center gap-2"
              >
                <img
                  alt="avatar-thuandmt"
                  src="/images/avatar/7.png"
                />
                <div
                  class="d-flex flex-column"
                >
                  <h6
                    class="mb-0"
                  >
                    Thuan Dao
                  </h6>
                  <span
                    class="text-secondary fs-7"
                  >
                    thuandmt
                  </span>
                </div>
              </div>
            </td>
            <td>
              6
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              2
            </th>
            <td>
              <div
                class="d-flex align-items-center gap-2"
              >
                <img
                  alt="avatar-sarahedo"
                  src="/images/avatar/1.png"
                />
                <div
                  class="d-flex flex-column"
                >
                  <h6
                    class="mb-0"
                  >
                    Sarah Edo
                  </h6>
                  <span
                    class="text-secondary fs-7"
                  >
                    sarahedo
                  </span>
                </div>
              </div>
            </td>
            <td>
              4
            </td>
            <td>
              2
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              3
            </th>
            <td>
              <div
                class="d-flex align-items-center gap-2"
              >
                <img
                  alt="avatar-mtsamis"
                  src="/images/avatar/3.png"
                />
                <div
                  class="d-flex flex-column"
                >
                  <h6
                    class="mb-0"
                  >
                    Mike Tsamis
                  </h6>
                  <span
                    class="text-secondary fs-7"
                  >
                    mtsamis
                  </span>
                </div>
              </div>
            </td>
            <td>
              3
            </td>
            <td>
              2
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              4
            </th>
            <td>
              <div
                class="d-flex align-items-center gap-2"
              >
                <img
                  alt="avatar-tylermcginnis"
                  src="/images/avatar/2.png"
                />
                <div
                  class="d-flex flex-column"
                >
                  <h6
                    class="mb-0"
                  >
                    Tyler McGinnis
                  </h6>
                  <span
                    class="text-secondary fs-7"
                  >
                    tylermcginnis
                  </span>
                </div>
              </div>
            </td>
            <td>
              2
            </td>
            <td>
              2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`PollList renders correctly PollList render as expected with props { items: [ [length]: 0 ] } 1`] = `
<body>
  <div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover table-bordered align-middle caption-top"
      >
        <caption>
          List of users
        </caption>
        <thead
          class="table-light"
        >
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              User
            </th>
            <th
              scope="col"
            >
              Answered
            </th>
            <th
              scope="col"
            >
              Asked
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</body>
`;
